src/XMonad/Layout.hs

```diff
@@ -155,6 +190,18 @@ instance Message NextNoWrap
 handle :: (LayoutClass l a, Message m) => l a -> m -> X (Maybe (l a))
 handle l m = handleMessage l (SomeMessage m)
 
+-- | Swap the two sub-layouts.
```

Comment:
This is misleading. I'd expect a `Choose l r a → Choose r l a`, but instead it's a `switch` that also happens to notify the inner layouts that they're going to be hidden.

Reply:
Oh and I just noticed that there's this `choose` function that could possibly replace both `swap` and `switch`, as it seems to be more general. IOW, `JumpToLayout` handling may be possible to implement concisely using `choose` alone.

Reply:
(If you want to test this, I expect that any decorated layout would fail to destroy its decoration windows if Hide messages aren't delivered to it on time.)